test/jtreg-ext/requires/VMProps.java

```diff
@@ -715,29 +715,77 @@ private String isFlagless() {
         return "" + result;
     }
 
-    private Stream<String> allFlags() {
+    private static Stream<String> allFlags() {
         return Stream.of((System.getProperty("test.vm.opts", "") + " " + System.getProperty("test.java.opts", "")).trim().split("\\s+"));
     }
 
+    public record Flag(String name, String value) {}
+
+    /**
+     * Split an -Xflag string into a name part (without leading dash)
+       and a value part ("true" if no value was given)
+     */
```

Comment:
May I suggest sparing the reader some effort and adding a couple of examples e.g.:
```
-Xmx4g => (Xmx, 4g)
-Xcheck:jni => (Xcheck, :jni)
-Xbootclasspath/a:<path> => (Xbootclasspath, /a:<path>)
```